### ISSUE TITLE
Possible compile error on Ubuntu 

### ISSUE THREAD

**Issue by ngsrinidhi** [tudelft3d/City4CFD#5]
There was a minor error during compilation in my Ubuntu 20.04. 

```
Error: 
City4CFD/src/io.cpp:227:80: error: invalid conversion from ‘OGRGeometry*’ to ‘OGRPolygon*’ [-fpermissive]
OGRPolygon *inputPolygon =inputMultipolygon->getGeometryRef(currentPolygon);
```

I found that the error is coming from pointer call in line 227 in io.cpp file. The problem goes away if we fix the pointer call to this: 
`OGRPolygon *inputPolygon=inputMultipolygon->getGeometryRef(currentPolygon)->toPolygon(); `

May be problem is specific to my machine/compiler. Just commenting here in case someone else faces the same issue.

**Comment by ipadjen** [comment]
Thanks for reporting. Do you use GDAL from the package manager or some other version?

Would be nice if to see if more people have this issue so we can pinpoint the cause.

**Comment by BW-Sampath** [comment]
I get the exact same error on Ubuntu 20.04 as well. I use GDAL from the package manager, which is version 3.0.4. I too confirm the error disappears when I change line 227 in io.cpp file to 
`OGRPolygon *inputPolygon=inputMultipolygon->getGeometryRef(currentPolygon)->toPolygon(); `

**Comment by ipadjen** [comment]
Thank you for confirming this @BW-Sampath.

**Comment by ipadjen** [comment]
Recreated the issue with a workflow and it seems fixed now with 99ace13. Thank you!

I will publish it as a new version, please let me know if there are still issues with it.

**Comment by ipadjen** [comment]
Closing this now. Please let me know if there are any other problems with this.
